src/strongbox/cli.clj

```diff
@@ -706,33 +706,34 @@
 
 (defn-spec refresh-user-catalogue-item nil?
   "refresh the details of an individual `addon` in the user catalogue, optionally writing the updated catalogue to file."
-  ([addon :addon/summary]
-   (refresh-user-catalogue-item addon true))
-  ([addon :addon/summary, write? boolean?]
-   (logging/with-addon addon
-     (info "refreshing details")
-     (try
-       (let [attempt-dry-run? false
-             refreshed-addon (find-addon (:url addon) attempt-dry-run?)]
-         (if refreshed-addon
-           (do (core/add-user-addon! refreshed-addon)
-               (when write?
-                 (core/write-user-catalogue!))
-               (info "... done!"))
-           (warn "failed to refresh catalogue entry")))
-       (catch Exception e
-         (error (format "an unexpected error happened while updating the details for '%s' in the user-catalogue: %s"
-                        (:name addon) (.getMessage e))))))))
+  [addon :addon/summary, db :addon/summary-list]
+  (logging/with-addon addon
+    (info "refreshing details")
+    (try
+      (let [{:keys [source source-id url]} addon
+            refreshed-addon (first (core/db-addon-by-source-and-source-id db source source-id))
```

Comment:
we should always expect this fn to return a single result. source+source-id should be unique.